src/adapters/chineseloc/chinese-source/chinese-source.js

```diff
@@ -0,0 +1,385 @@
+      parsedLength = parsedLength + rawElement.length + 2
+
+      let parsedElement = rawElement
+      let word = parsedElement.substr(0, parsedElement.indexOf('[')).trim()
+      parsedElement = parsedElement.substr(parsedElement.indexOf('[') + 1)
+
+      let pinyin = parsedElement.substr(0, parsedElement.indexOf(']'))
+      parsedElement = parsedElement.substr(parsedElement.indexOf(']') + 1)
+
+      let shortDef = parsedElement.trim().substr(1)
+      shortDef = shortDef.substr(0, shortDef.length - 1).replace(/\//g, '; ')
+
+      formattedData.set(currentIndex, {
+        word, pinyin, shortDef
+      })
+    })
+
+    return formattedData
+  }
+
+  static convertIDX (rawData) {
+    let formattedData = new Map()
+
+    // let data = rawData.slice(40, 50)
+    rawData.forEach(rawElement => {
+      let codes = []
+      let checkEl = formattedData.get(rawElement[0])
+
+      // console.info('checkEl - ', checkEl)
+
+      if (checkEl) {
+        // console.info('checkEl codes before - ', codes)
+        codes.push(checkEl.codes[0])
+        // console.info('checkEl codes after - ', codes)
+      }
+
+      codes.push(rawElement[1])
+      formattedData.set(rawElement[0], {
+        word: rawElement[0],
+        codes: codes
+      })
+    })
+
+    // console.info('formattedData - ', formattedData)
+    return formattedData
+  }
+
+  static convertHanzi (rawData) {
+    let formattedData = new Map()
+
+    // let data = rawData.slice(0, 10)
+    rawData.forEach(rawElement => {
+      let character = rawElement[0]
+      let typeCh = rawElement[1]
+      let typeText = rawElement[2]
+
+      let element = {
+        character
+      }
+      element[typeCh] = typeText
+
+      let prevElement = formattedData.get(rawElement[0])
+
+      if (prevElement) {
+        Object.keys(prevElement).filter(key => key !== 'character').forEach(key => {
+          element[key] = prevElement[key]
+        })
+      }
+      formattedData.set(character, element)
+    })
+    return formattedData
+    // console.info('formattedData - ', formattedData)
+  }
+
+  static convertRadicals (rawData) {
+
+  }
+
+  static findWord (targetWord, wordIDX, wordDict) {
+    let searchedIdxElement = wordIDX.get(targetWord)
+
+    if (searchedIdxElement) {
+      // console.info('serchedIdxElement.codes - ', searchedIdxElement.codes)
+      if (searchedIdxElement.codes && searchedIdxElement.codes.length > 0) {
+        // console.info('serchedIdxElement.codes - ', searchedIdxElement.codes)
+        return searchedIdxElement.codes.map(code => {
+          // console.info('findWord code - ', code, wordDict.get(code))
+          return wordDict.get(code)
+        })
+      }
+    }
+
+    return null
+  }
+
+  static lookupChinese (targetWord) {
+    // console.info('lookupChinese dHanziDict - ', dHanziDict.size)
+    let cpWord = targetWord
+    let count = 0
+    let format = 'simp'
+
+    let rs = []
+
+    let result
+    while (cpWord.length > 1) {

```

Comment:
I found, why I have this check here.
This check is from the Apheios 1.0 - 
```
       // for each initial substring of word
        // For now, don't check single characters - use hanzi data instead
        // To enable single-char dictionary entries,
        // use test "cpWord.length > 0"
        while (cpWord.length > 1)
        {
```

Reply:
Let's change it :)

Reply:
I have changed it - and now it could translate all the words from my examples!

Reply:
I see -- from the comment it looks like the Alpheios 1.0 code used the hanzi.dat file to retrieve the single character entries.  I am not sure why we did that exactly --- maybe the cedict data was incomplete in this regard. I do think that it could be useful to retain a fall back to retrieve single character data from the unihan database file if it isn't found in the cedict file but another option might be to merge these two files from the start when we generate the json version of cedict. @kirlat what do you think?

Reply:
I'm not sure if we can merge the unihan database with cedict because they may present different data and have different record formats. But if we'll find a way to do that it will be great. 
If we won't then I agree with @balmas suggestion to look up cedict first and then query the unihan db. We can load and handle the latter the same way we do handle the cedict.

Reply:
either way we need to query unihan for the radical, stroke count and pronunciation. For the prototype the way this is now is fine.  For the full implementation we'll have to decide whether to combine these data sources upfront, or keep them separate.